File: android/src/main/java/top/huic/tencent_im_plugin/message/GroupTipsMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMGroupTipsElem;

import top.huic.tencent_im_plugin.message.entity.GroupTipsMessageEntity;

/**
 * 群提示消息节点
 */
public class GroupTipsMessageNode extends AbstractMessageNode<V2TIMGroupTipsElem, GroupTipsMessageEntity> {
    @Override
    public String getNote(V2TIMGroupTipsElem elem) {
        return "[群提示]";
    }

    @Override
    public GroupTipsMessageEntity analysis(V2TIMGroupTipsElem elem) {
        return new GroupTipsMessageEntity(elem);
    }

    @Override
    public Class<GroupTipsMessageEntity> getEntityClass() {
        return GroupTipsMessageEntity.class;
    }
}

File: ios/Classes/message/GroupTipsMessageNode.swift
import ImSDK

//
//  GroupTipsMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/21.
//  群提示消息节点
public class GroupTipsMessageNode: AbstractMessageNode {
    override func getNote(elem: V2TIMElem) -> String {
        "[群提示]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        GroupTipsMessageEntity(elem: elem as! V2TIMGroupTipsElem);
    }
}
